pkg/annotations/annotation.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package annotations makes it possible to track use of resource annotations within a procress
+// in order to generate documentation for these uses.
+package annotations
+
+import (
+	"sort"
+	"sync"
+
+	"istio.io/istio/pkg/log"
+)
+
+// Annotation describes a single resource annotation
+type Annotation struct {
+	// The name of the annotation.
+	Name string
+
+	// Description of the annotation.
+	Description string
+
+	// Hide the existence of this annotation when outputting usage information.
+	Hidden bool
+
+	// Mark this annotation as deprecated when generating usage information.
+	Deprecated bool
+}
+
+var allAnnotations = make(map[string]Annotation)
+var mutex sync.Mutex
+
+// Returns a description of this process' annotations, sorted by name.
+func AnnotationDescriptions() []Annotation {
```

Comment:
Just Descriptions? It will read as ```annotation.Descriptions()``` in the calling code.

Reply:
Done